docs/doc/10-faq/06-how-to-write-scalar-functions.md

```diff
@@ -128,66 +60,167 @@ Would look like this:
 
 ```
 
-#### Constant column
+In most cases, we can ignore null for simd operation, and add the null mask to the result after the operation.
+This is very common optimization and widely used in arrow's compute system.
 
-Sometimes column is constant in the block, such as: `select 3 from table`, the column 3 is always 3, so we can use a constant column to represent it. This is useful to save the memory during computation.
+### Special column
 
-So databend's DataColumn is represented by:
+-  Constant column
 
-```rust
-pub enum DataColumn {
-    // Array of values. Series is wrap of arrow's array
-    Array(Series),
-    // A Single value.
-    Constant(DataValue, usize),
+    Sometimes column is constant in the block, such as: `select 3 from table`, the column 3 is always 3, so we can use a constant column to represent it. This is useful to save the memory space during computation.
+
+    So databend's DataColumn is represented by:
+
+    ```rust
+    #[derive(Clone)]
+    pub struct ConstColumn {
+        length: usize,
+        column: ColumnRef,
+    }
+    ```
+- Nullable column
+
+    By defaults, column are not nullable. If we want nullable column, we can use this to represent it.
```

Comment:
typo to:
By default, columns are not nullable. If we want a nullable column, we can use this to represent it.